--- APKBUILD.orig
+++ APKBUILD
@@ -13,6 +13,7 @@
 	ntfs-3g-dev
 	f2fs-tools-dev
 	hfsprogs
+	xfsprogs-dev
 	ncurses-dev
 	automake
 	autoconf
@@ -29,6 +30,8 @@
 subpackages="$pkgname-doc $pkgname-lang"
 source="https://github.com/Thomas-Tsai/partclone/archive/$pkgver/partclone-$pkgver.tar.gz
 	musl.patch
+	radix-tree.patch
+	linux.patch
 	"
 
 if [ "$CARCH" = "riscv64" ]; then
@@ -48,6 +51,7 @@
 		--enable-btrfs \
 		--enable-minix \
 		--enable-f2fs \
+		--enable-xfs \
 		--enable-hfsp
 	make
 }
@@ -63,4 +67,6 @@
 sha512sums="
 32dced369973c39412c562ab8731ff7b490f464b5ed4033456e6b23f3f79c437f43dc9db4c79b4821f35e4997d1444c8d80c53816b2a3bb90836fb5ad33b368e  partclone-0.3.20.tar.gz
 0676b34c5818f8866e733b0aee151e8220b8db839ba732acfdaad4dcea81a4b5d514a520e43c79efcf53ba0788c9a7f06c043441b4d1f56dd684882329efce9d  musl.patch
+703aa4eade4c98951bb23133b9853250726a8db4b8d480a61901a29280397376ef0d78573b8b626f2625413b429866061c5eeaa740e8e473b28ce50322632772  radix-tree.patch
+363d87c17c7d94508a1d4ae4f1f0911a96ed3f6c5bdc3b579587e360bfb3bfd713a82994dad91406acd645a8673bbac9e26f823df6d129631da9eb42baf3f479  linux.patch
 "
